File: globalAndLocalInversions.go
// problem: problem: https: // leetcode.com/problems/global- and -local-inversions/
// Runtime: 44 ms, faster than 84.62% of Go online submissions for Global and Local Inversions.
// Memory Usage: 6.4 MB, less than 76.92% of Go online submissions for Global and Local Inversions.

package algorithms

func isIdealPermutation(A []int) bool {
	for i, val := range A {
		tmp := val - i
		if tmp > 1 || tmp < -1 {
			return false
		}
	}
	return true
}

File: globalAndLocalInversions.py
# problem: https: // leetcode.com/problems/global- and -local-inversions/
# Runtime: 316 ms, faster than 97.70 % of Python3 online submissions for Global and Local Inversions.
# Memory Usage: 15.1 MB, less than 44.06 % of Python3 online submissions for Global and Local Inversions.

from typing import List


class Solution:
    def isIdealPermutation(self, A: List[int]) -> bool:
        for i in range(len(A)):
            if abs(i - A[i]) > 1:
                return False
        return True
